### DISCUSSION TITLE
v0.1.4

### DISCUSSION THREAD

**Discussion by bug-or-feature** [bug-or-feature/bc-utils#11] [Announcements]
## v0.1.4 (2024-02-15)

* now downloads hourly and daily prices to separate files, eg Day_GOLD_20230200.csv, Hour_GOLD_20230400.csv
* better logic for handling periods with no or low data
* new script to update previously downloaded files
* new script to separate files from previous versions into split frequency files
* example code
* tests against Python 3.10
* black

This new release is a big re-write of the existing code. The main problem with the previous version was that the `tick_date` values supplied by Barchart were inaccurate. We used that date to work out when to switch from hourly to daily data. Because the dates were inaccurate, we would end up with lots of files with bad data: empty files, files with only a couple of rows, invalid price rows, files with mixed frequency rows. It seemed to be worse for more obscure contracts. It made it very fiddly to stitch the contract files together, requiring lots of manual edits around the switchover date.

The new version resolves all those problems by downloading daily and hourly prices separately. Barchart have increased the daily download allowance to 250 files for the premier account, so we're no worse off. There is much better logic for detecting bad data, so we don't end up with invalid files. There is a script to convert files downloaded with a previous version, to the new split frequency format.

**Comment by littlebigfis** [comment]
Awesome! feeling Grateful for how much time this is saving me.
